File: RetailChainUi/src/services/dashboard.service.js
import { axiosPrivate } from "./api/axiosClient";

const dashboardService = {
  getSummary: async (timeRange = "30days") => {
    const response = await axiosPrivate.get("/dashboard/summary", {
      params: { timeRange },
    });
    return response?.data || null;
  },
};

export default dashboardService;

File: RetailChainService/src/main/java/com/sba301/retailmanagement/service/DashboardService.java
package com.sba301.retailmanagement.service;

import com.sba301.retailmanagement.dto.response.DashboardSummaryResponse;

public interface DashboardService {
    /**
     * Tổng hợp dữ liệu dashboard theo timeRange.
     * timeRange: "30days" | "quarter" | "ytd"
     */
    DashboardSummaryResponse getSummary(String timeRange);
}

